packages/gitbook/src/components/TableOfContents/TableOfContents.tsx

```diff
@@ -1,21 +1,23 @@
-import { SiteInsightsTrademarkPlacement } from '@gitbook/api';
-import type { GitBookSiteContext } from '@v2/lib/context';
+'use client';
+
 import type React from 'react';
 
+import { MobileMenuSheet } from '@/components/MobileMenu';
+import { useIsMobile } from '@/hooks/useIsMobile';
 import { tcls } from '@/lib/tailwind';
-
-import { PagesList } from './PagesList';
-import { TOCScrollContainer } from './TOCScroller';
 import { TableOfContentsScript } from './TableOfContentsScript';
-import { Trademark } from './Trademark';
 
 export function TableOfContents(props: {
-    context: GitBookSiteContext;
     header?: React.ReactNode; // Displayed outside the scrollable TOC as a sticky header
-    innerHeader?: React.ReactNode; // Displayed outside the scrollable TOC, directly above the page list
+    children: React.ReactNode;
 }) {
-    const { innerHeader, context, header } = props;
-    const { space, customization, pages } = context;
+    const { header, children } = props;
+    const isMobile = useIsMobile();
+
+    // If the screen is mobile, we use the mobile menu sheet to display the table of contents.
+    if (isMobile) {
+        return <MobileMenuSheet>{children}</MobileMenuSheet>;
+    }
```

Comment:
Why do we need that. I don't understand why we don't keep the same element and we just switch its position to absolute or something like that? In the app, in mobile I use another layout but we could keep the same component like in desktop actually (I think I will refactor it at some point). I think we could do something smarter here.